### PR TITLE
Implement ACL support

### PR DESCRIPTION
**Pull request by mgerstner**
This addresses the ACL feature request we received.

Along with the feature addition also comes a bit of refactoring here and there, where it became obvious that the code is no longer simple enough.

Also this updates and extends the documentation on a general level, as well as specifically regarding to the new ACL feature.

Also the test framework has been extended to cover the new ACL features.

### CODE REVIEW

**Comment by FilippoBonazziSUSE** [comment]
Reference: https://jira.suse.com/browse/PED-6816

File: src/profparser.cpp [hunk 1]
```diff
@@ -71,7 +68,7 @@ void ProfileParser::parse(const std::string &path, std::ifstream &fs) {
     }
 }
 
-void ProfileParser::parseExtraLine(const std::string &line, std::function<void(const std::string_view)> printBadLine) {
```
Comment (FilippoBonazziSUSE) [inline src/profparser.cpp]: Do you still need `#include <functional>` now in `profparser.h` then?

**Review by FilippoBonazziSUSE — APPROVED** [review]
Very nice work.

LGTM other than possibly one comment.

File: src/profparser.cpp (hunk 1 above)
Comment (mgerstner) [reply src/profparser.cpp]: Indeed. I checked all includes and found another unused one. Pushed a commit to clean this up.

**Review by FilippoBonazziSUSE — APPROVED** [review]
Alright, lgtm. Good to merge

**Pull request merged**